app/src/main/java/de/westnordost/streetcomplete/quests/crossing_type/AddCrossingTypeForm.java

```diff
@@ -0,0 +1,43 @@
+package de.westnordost.streetcomplete.quests.crossing_type;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
+
+public class AddCrossingTypeForm extends ImageListQuestAnswerFragment
+{
+    private final OsmItem[] TYPES = new OsmItem[] {
+            new OsmItem("traffic_signals", R.drawable.crossing_type_signals, R.string.quest_crossing_type_signals),
+            new OsmItem("uncontrolled", R.drawable.crossing_type_zebra, R.string.quest_crossing_type_uncontrolled),
+            new OsmItem("island", R.drawable.crossing_type_island, R.string.quest_crossing_type_island),
+            new OsmItem("unmarked", R.drawable.crossing_type_unmarked, R.string.quest_crossing_type_unmarked)
+    };
+
+    @Override protected int getMaxSelectableItems()
+    {
+        return -1;
```

Comment:
This allows the user to input invalid data, as traffic signals, uncontrolled and unmarked are mutually exclusive. This is not good, because it can be avoided.

The `island` tag and in particular that it can be combined with any other of these values but the other values cannot be combined with each other appears somewhat problematic to me. This calls for ("schreit nach") `island` should really be a separate tag. [It has also been noted on the discussion page](https://wiki.openstreetmap.org/wiki/Talk:Key:crossing#Island_.2Aor.2A_Traffic_signals_.3F). Also note that there is a big warning at the start of the page _"This page is being considered for cleanup. Please discuss this page."_, so I'd say at least there is agreement on that the wiki page on the crossing tag is problematic.

0.25% of all `crossing=`-tags use multiple values (with `island`). This is really insignificant.

In my opinion, there are two ways to solve this:
1. Since 0.25% is _really_ insignificant, and given that that tagging scheme is problematic and discussed anyway, simply do not implement this multi-choice. `island` then is to be interpreted as "no traffic lights, no markings, but... an island". This is how I imagine this tag was originally conceived for before other osm-wiki-people found that it could be combined with other values as well.
2. Do not show this option at all because it is not clear anyway what a simple crossing=island tag would imply (Does it imply `crossing=unmarked`? Not defined). Unlike the other three values (uncontrolled/unmarked/with traffic lights ) It is of no use for routing applications.
Then, maaaybe there can be a second quest which asks whether there is an island for this crossing or not which tags `crossing:island=yes/no`. But, to be honest, the value of specifying whether there is an island or not is completely oblivious to me. (And also `crossing:island` is undocumented)